mathics/builtin/attributes.py

```diff
@@ -466,9 +466,14 @@ class OneIdentity(Predefined):
 
     'OneIdentity' affects pattern matching:
     >> SetAttributes[f, OneIdentity]
-    >> a /. f[args___] -> {args}
+    >> a /. f[x_:0, u_] -> {u}
      = {a}
-    It does not affect evaluation:
+
+    However, without a default argument, the pattern does not match:
+    >> a /. f[u_] -> {u}
+     = a
+
+    Also, it does not affect evaluation:
```

Comment:
I find these examples and the documentation to be extremely confusing. 

Do you mind if I just rewrite the examples and doc here? 

A small change is to replace "Also, it does not" with "'OneIdentity' does not" 

Page 272 of the Mathematica 5 book motivates OneIdentity: 

> ... in the case where `x_` matches a single argument in a flat function, the question comes up as to whether the object it matches is really just and argument _a_ itself or _f[a]_. Mathematica choses the first of these cases if the function caries the attribute `OneIdentity` and chooses the second case otherwise.  

"In" 16-20" make this clear.

Reply:
Sure, go ahead.